src/adapters/azure/index.ts

```diff
@@ -7,22 +7,29 @@ import { getHandleUpload } from './handleUpload'
 import { extendWebpackConfig } from './webpack'
 
 export interface Args {
```

Comment:
The args need to require a connectionString or a storageClient. I wouldn't mark both as optional.
You need something like:

```
type Args = {
  containerName: string
  baseURL: string
  allowContainerCreate: boolean
} & ( {
  connectionString: string
  storageClient?: never
 } | { 
  storageClient: ContainerClient
  connectionString?: never
 })
```